### DISCUSSION TITLE
How to run the batch delete apex classes?

### DISCUSSION THREAD

**Discussion by baobao917** [jongpie/NebulaLogger#638] [Q&A] [answered]
We have Nebula logger records taking up an inordinate amount of data storage.  How can we run both of the below apex classes to delete the existing records and run the scheduled job to delete records?  Am on release 4.11.7

![image](https://github.com/jongpie/NebulaLogger/assets/104730171/00c66bb9-df0b-4828-8559-8033e36b745f)

**Comment by jongpie** [comment, marked as answer]
Hi @baobao917 I think the easiest option is to schedule the `LogBatchPurgerScheduler` class to run daily under `Setup`. You can do that with these steps:
1. Under `Setup`, go to `Apex Classes` and click the button `Schedule Apex`
  
  ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/a63a858b-405d-49ef-acf6-734d3a690e9d)

2. On the `Schedule Apex` page, provide a `Job Name`, select `LogBatchPurgeScheduler` for `Apex Class`, and choose the frequency options you'd like
  
  ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/5c21d00f-9a20-4cf6-bd01-fc44c2b7e9fe)

Once you've done that, the scheduled job will automatically delete any old data in `Log__c`, `LogEntry__c` and `LogEntryTag__c`, based on the date field `Log__c.LogRetentionDate__c`.
